### PR TITLE
Flip grpc client cache flag

### PR DESCRIPTION
**Pull request by chaoqin-li1123**
Commit Message: Make grpc client cache default enable.
Additional Description:
Risk Level: low
Testing: change testing to verify new behavior
Docs Changes: document the behavior change in current.rst
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by jmarantz — COMMENTED** [review]
Code looks fine; I thought we were going to enable this at one call-site per PR though.

Or is there only one remaining call-site at this point?

/wait

**Comment by chaoqin-li1123** [comment]
Per [our discussion](https://github.com/envoyproxy/envoy/pull/18486#pullrequestreview-781120352), it is ok to flip a runtime even when involves many components and filters.

File: docs/root/version_history/current.rst
Comment (alyssawilk) [inline docs/root/version_history/current.rst:17]: Can you make sure comment what the functional change is?

Also this got lost as well due to the waiting tag, sorry.  I've removed it so hopefully it will make progress again, but you'll want to main merge for current.rst

**Comment by jmarantz** [comment]
also needs main merge

**Comment by chaoqin-li1123** [comment]
./retest

**Review by jmarantz — APPROVED** [review]
Needs a non-Google senior maintainer approval:

/assign-from @envoyproxy/senior-maintainers

**Review by zuercher — APPROVED** [review]

**Pull request merged**
